### ISSUE TITLE
AutoHeight with unknown maxHeight?

### ISSUE THREAD

**Issue by TheOpti** [malte-wessel/react-custom-scrollbars#125]
Right now autoHeightMin, autoHeightMax take parameter of type number, which means number of pixels. How can I pass something like 'calc(100vh - 120px)' as my max height in scroll container?

Is it possible right now?

**Comment by Dmitry-N-Medvedev** [comment]
Not 100% sure, but it might be as folloing:

```
autoHeightMin={`calc(100vh - 120px)`}
```

**Comment by KyleAMathews** [comment]
I can confirm @Dmitry-N-Medvedev's solution works. Would be good to add to docs.

**Comment by sebhewelt** [comment]
Yeah, it works, but what if I would like to have the height to be 100% of the container?

 - Setting it ` autoHeightMin={100%} , makes it about 200px high, 
- setting it ` autoHeightMin={`calc(100vh)`}` disables scroll events, because height than, is indeed 100vh, not fitting the container.

**Comment by stevenmusumeche** [comment]
same problem here, i'd like it to be 100% height of it's container but nothing that I use works

**Comment by sag1v** [comment]
well i was hopping there was a way to set a real auto height that fit to the container or the first positioned relative parent. No luck so far.

**Comment by hongbo-miao** [comment]
I found
```html
<Scrollbars style={{ height: '100%' }}></Scrollbars>
```
does not work.

But
```html
<Scrollbars autoHeight autoHeightMin="100%" autoHeightMax="100%"></Scrollbars>
```
works for me!

**Comment by drewlustro** [comment]
Also having this issue... unknown `maxHeight` 🤷‍♂️

**Comment by team-moonrise** [comment]
Had the opposite effect that @Hongbo-Miao had. 
`style={{ height: '100%' }}`

Worked great for me, where autoHeight, didn't work. Not quite sure why.

**Comment by rocketraman** [comment]
Hmm... `style={{ height: '100%' }}` worked for me as well... I'm not positive, but I think it really depends on how your parent container elements are defined. If your parent element defines it's height according to it's content, then you'll probably need to use `autoHeight` somehow. On the other hand, if your parent element defines it's height according to it's own space available, then `style={{ height: '100%' }}` will probably be the way to go.

In my case, I had a flexbox column container with setting `auto`. This was not working, apparently because of the `auto` flex-basis, which sets the initial height of the flex container to the size of its content -- the result was a 0 height scrollbar and content. Changing this to flex `1 1 0` (same as auto, but with a non-content based `flex-basis`) fixed the problem.

It's likely this just needs some documentation and some samples to know how to use it properly in various situations.

**Comment by ikorgik** [comment]
Good point @rocketraman!
`style={{ height: '100%' }}` works well to me when I set `height: '100%'` for the parent container.

**Comment by kiennguyen278** [comment]
i use
`<Scrollbars
                                    autoHeight
                                    autoHeightMin={100}
                                    autoHeightMax="calc(80vh - 36px - 35px - 75px)">
content scroll
</Scrollbars>
it work for me
`

**Comment by vladi-strilets** [comment]
Hi everyone, I had the same issue. 

As my parent has `display: flex; height: 100%` I  use `<Scrollbars style={{ flex: 1 }}>{children}</Scrollbars>`

It works like a charm ;)
